operator/src/main/java/oracle/kubernetes/operator/helpers/JobHelper.java

```diff
@@ -253,12 +253,18 @@ private String jobDescription(@Nonnull V1Job job) {
                          + ", started at " + job.getMetadata().getCreationTimestamp();
       }
 
-      private boolean isInProgressJobOutdated(V1Job job) {
+      private boolean isInProgressJobOutdated(Packet packet, V1Job job) {
         return Optional.ofNullable(job)
-            .map(j -> hasNotCompleted(j) && (hasAnyImageChanged(j) || hasIntrospectVersionChanged(j)))
+            .map(j -> hasNotCompleted(j) && isJobOutdated(packet, j))
             .orElse(false);
       }
 
+      private boolean isJobOutdated(Packet packet, @Nonnull V1Job job) {
+        return hasAnyImageChanged(job)
+            || hasIntrospectVersionChanged(job)
+            || isInspectionRequiredAndGenerationChanged(packet, job);
```

Comment:
My understanding of the requirements was that we wanted to wait for the existing/running introspection job to finish unless the `image`, `auxiliaryImages`, `introspectVersion` or `restartVersion` field in the domain resource change. If the MII domain resource is patched with a change that would cause introspection to fail, can the test also patch the `introspectVersion` to abort the current introspection and trigger a new introspection? I'm not sure if the requirements have changed recently or if I'm missing something.